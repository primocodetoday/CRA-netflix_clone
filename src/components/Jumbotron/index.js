import React from 'react';
import PropTypes from 'prop-types';
import {
  Item,
  Inner,
  Container,
  Pane,
  Title,
  SubTitle,
  Image,
} from './styles/jumbotron';

const Jumbotron = ({ children, direction }, ...restProps) => {
  return (
    <Item {...restProps} direction={direction}>
      <Inner>{children}</Inner>
    </Item>
  );
};

Jumbotron.Container = function JumbotronContainer({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
};

Jumbotron.Pane = function JumbotronPane({ children, ...restProps }) {
  return <Pane {...restProps}>{children}</Pane>;
};

Jumbotron.Title = function JumbotronTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

Jumbotron.SubTitle = function JumboSubtronTitle({ children, ...restProps }) {
  return <SubTitle {...restProps}>{children}</SubTitle>;
};

Jumbotron.Image = function JumbotronImage({ ...restProps }) {
  return <Image {...restProps} />;
};

Jumbotron.propTypes = {
  direction: PropTypes.string,
};

Jumbotron.defaultProps = {
  direction: 'row',
};

export default Jumbotron;
