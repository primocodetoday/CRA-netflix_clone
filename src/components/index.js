export { default as Footer } from 'components/Footer';
export { default as Accordion } from 'components/Accordion';
export { default as Jumbotron } from 'components/Jumbotron';
