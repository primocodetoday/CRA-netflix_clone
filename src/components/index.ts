export { Accordion } from 'components/Accordion/Accordion';
export { Banner } from 'components/Banner/Banner';
export { Card } from 'components/Card/Card';
export { CTAForm } from 'components/CTAForm/CTAForm';
export { Form } from 'components/Form/Form';
export { Footer } from 'components/Footer/Footer';
export { Header } from 'components/Header/Header';
export { Jumbotron } from 'components/Jumbotron/Jumbotron';
export { Loading } from 'components/Loading/Loading';
export { Player } from 'components/Player/Player';
export { Profiles } from 'components/Profiles/Profiles';
