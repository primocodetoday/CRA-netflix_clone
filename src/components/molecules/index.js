export { default as Footer } from 'components/molecules/Footer/Footer';
export { default as Accordion } from 'components/molecules/Accordion/Accordion';
export { default as Jumbotron } from 'components/molecules/Jumbotron/Jumbotron';
export { default as CTAForm } from 'components/molecules/CTAForm/CTAForm';
export { default as Header } from 'components/molecules/Header/Header';
