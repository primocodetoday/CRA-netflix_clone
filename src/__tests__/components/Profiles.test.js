import React from 'react';
import { render, screen } from '@testing-library/react';
import { Profiles } from 'components';

describe('<Profiles/> renders properly', () => {
  const user = {
    displayName: 'Sebastian',
  };

  it('with data', () => {
    const { container } = render(
      <Profiles>
        <Profiles.Title>Who&apos;s watching?</Profiles.Title>
        <Profiles.List>
          <Profiles.Item
            onClick={() => {
              return console.log(`I'm working`);
            }}
          >
            <Profiles.Avatar src="assets/images/fake.png" data-testid="profiles-avatar" />
            <Profiles.Avatar data-testid="profiles-loading" />
            <Profiles.Name>{user.displayName}</Profiles.Name>
          </Profiles.Item>
        </Profiles.List>
      </Profiles>,
    );
    expect(screen.getByText("Who's watching?")).toBeTruthy();
    expect(screen.getByText('Sebastian')).toBeTruthy();
    expect(screen.getAllByTestId('profiles-avatar')).toBeTruthy();
    expect(screen.getAllByTestId('profiles-loading')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
