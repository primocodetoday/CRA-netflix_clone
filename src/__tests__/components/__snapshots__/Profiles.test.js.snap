// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profiles/> renders properly with data 1`] = `
<div
  class="StyledProfiles__Container-sc-5dr50d-0 UjZnw"
>
  <h1
    class="StyledProfiles__Title-sc-5dr50d-4 iIRrSj"
  >
    Who's watching?
  </h1>
  <ul
    class="StyledProfiles__List-sc-5dr50d-2 gcuNwP"
  >
    <li
      class="StyledProfiles__Item-sc-5dr50d-5 jvRLON"
    >
      <img
        class="StyledProfiles__Avatar-sc-5dr50d-1 kMpxeQ"
        data-testid="profiles-avatar"
        src="assets/images/profiles/assets/images/fake.png.png"
      />
      <img
        class="StyledProfiles__Avatar-sc-5dr50d-1 kMpxeQ"
        data-testid="profiles-loading"
        src="assets/images/loading.gif"
      />
      <p
        class="StyledProfiles__Name-sc-5dr50d-3 fFOHqO"
      >
        Sebastian
      </p>
    </li>
  </ul>
</div>
`;
